runtime/binding-tcp/src/main/java/io/aklivity/zilla/runtime/binding/tcp/internal/TcpBindingContext.java

```diff
@@ -41,7 +40,8 @@ final class TcpBindingContext implements BindingContext
         TcpConfiguration config,
         EngineContext context)
     {
-        MutableInteger capacity = new MutableInteger(ENGINE_WORKER_CAPACITY.getAsInt(config));
+        LongConsumer capacityUsage = context.supplyUtilizationMetric();
+        CapacityTracker capacity = new CapacityTracker(ENGINE_WORKER_CAPACITY.getAsInt(config), capacityUsage);
```

Comment:
Suggest passing `config` and `context` to `TcpCapacityTracker` and handling the metric lookup and the capacity lookup in the constructor.